analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/RestrictDeserializedTypesBase.cs

```diff
@@ -26,4 +29,169 @@ public abstract class RestrictDeserializedTypesBase : SymbolicRuleCheck
     protected const string MessageFormat = "{0}";
     private const string RestrictTypesMessage = "Restrict types of objects allowed to be deserialized.";
     private const string VerifyMacMessage = "Serialized data signature (MAC) should be verified.";
+
+    private static readonly KnownType[] FormattersWithBinder = new[]
+    {
+        KnownType.System_Runtime_Serialization_Formatters_Binary_BinaryFormatter,
+        KnownType.System_Runtime_Serialization_NetDataContractSerializer,
+        KnownType.System_Runtime_Serialization_Formatters_Soap_SoapFormatter
+    };
+    private static readonly KnownType JavaScriptSerializer = KnownType.System_Web_Script_Serialization_JavaScriptSerializer;
+    private static readonly KnownType LosFormatter = KnownType.System_Web_UI_LosFormatter;
+    private static readonly KnownType[] TypesWithDeserializeMethod = FormattersWithBinder.Append(JavaScriptSerializer).ToArray();
+
+    private readonly Dictionary<ISymbol, SyntaxNode> additionalLocationsForSymbols = new();
+    private readonly Dictionary<IOperation, SyntaxNode> additionalLocationsForOperations = new();
+
+    protected abstract bool IsBindToType(SyntaxNode methodDeclaration);
```

Comment:
At first read the `IsXXXType` name implies we're looking for a type instead of a method.
`IsBindToType` => `IsBindToTypeMethod`
`IsResolveType` => `IsResolveTypeMethod`